pkg/endpoint/restore.go

```diff
@@ -88,3 +95,135 @@ func ReadEPsFromDirNames(owner regeneration.Owner, basePath string, eptsDirNames
 	}
 	return possibleEPs
 }
+
+// toRestoredEndpoint converts the Endpoint to its corresponding
+// restoredEndpoint, which contains all of the fields that are needed upon
+// restoring an Endpoint after cilium-agent restarts.
+func (e *Endpoint) toRestoredEndpoint() *restoredEndpoint {
+
+	return &restoredEndpoint{
+		ID:                    e.ID,
+		ContainerName:         e.ContainerName,
+		ContainerID:           e.ContainerID,
+		DockerNetworkID:       e.DockerNetworkID,
+		DockerEndpointID:      e.DockerEndpointID,
+		DatapathMapID:         e.DatapathMapID,
+		IfName:                e.IfName,
+		IfIndex:               e.IfIndex,
+		OpLabels:              e.OpLabels,
+		LXCMAC:                e.LXCMAC,
+		IPv6:                  e.IPv6,
+		IPv4:                  e.IPv4,
+		NodeMAC:               e.NodeMAC,
+		SecurityIdentity:      e.SecurityIdentity,
+		Options:               e.Options,
+		DNSHistory:            e.DNSHistory,
+		K8sPodName:            e.K8sPodName,
+		K8sNamespace:          e.K8sNamespace,
+		DatapathConfiguration: e.DatapathConfiguration,
+	}
+}
+
+// restoredEndpoint contains the fields from an Endpoint which are needed to be
+// restored if cilium-agent restarts.
+//
+//
+// WARNING - STABLE API
+// This structure is written as JSON to StateDir/{ID}/lxc_config.h to allow to
+// restore endpoints when the agent is being restarted. The restore operation
+// will read the file and re-create all endpoints with all fields which are not
+// marked as private to JSON marshal. Do NOT modify this structure in ways which
+// is not JSON forward compatible.
+//
+type restoredEndpoint struct {
```

Comment:
`serializableEndpoint` might better capture the intent here. `restoredEndpoint` might imply that it's already restored rather than the endpoint being restored. Not a big deal either way.

Reply:
Done!